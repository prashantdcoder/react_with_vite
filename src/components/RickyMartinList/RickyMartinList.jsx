/* eslint-disable no-unused-vars */
import PropTypes from "prop-types";
import React from "react";

import RickyMartinListItem from "../RickyMartinListItem/RickyMartinListItem";

const RickyMartinList = ({ list }) => {
  const listItems = list && list.map((item) => (
    <RickyMartinListItem key={item.id} data={item} />
  ));
  return <div className="list-container">{listItems}</div>;
};
RickyMartinList.propTypes = {
  list: PropTypes.array,
};
export default RickyMartinList;
