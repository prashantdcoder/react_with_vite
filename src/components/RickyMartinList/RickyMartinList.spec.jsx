/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import RickyMartinList from "./RickyMartinList";

describe("Component", () => {
  it("should render list correctly", () => {
    const mockList = [
      {
        id: 1,
        name: "Prashant",
        status: "ALIVE",
        gender: "male",
        episode: 56,
        image: "test",
        location: {name: "bangalore"},
        origin: {name: "bangalore"},
      },
    ];
    const { asFragment } = render(<RickyMartinList list={mockList} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
