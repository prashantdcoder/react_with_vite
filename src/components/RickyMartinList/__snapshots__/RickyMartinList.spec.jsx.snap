// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should render list correctly 1`] = `
<DocumentFragment>
  <div
    class="list-container"
  >
    <div
      class="ricky-martin-list-item-container"
    >
      <div
        class="item-group-container d-flex justify-space-btwn"
      >
        <span
          class="status-badge-container bg-green"
        >
          ALIVE
        </span>
        <span
          style="font-size: 20px;"
        >
          ðŸ‘¨
        </span>
      </div>
      <div
        class="item-group-container d-flex justify-center"
      >
        <div
          class="avatar-container"
        >
          <img
            alt="avatar_img"
            loading="lazy"
            src="test"
          />
          ;
        </div>
      </div>
      <div
        class="item-group-container d-flex justify-center flex-col align-center profile-container"
      >
        <h3
          class="profile-item"
        >
          Prashant
        </h3>
        <span
          class="profile-item"
        >
          bangalore
        </span>
        <span
          class="profile-item"
        >
          bangalore
        </span>
      </div>
      <div
        class="item-group-container meta-container"
      >
        <span>
          Episode
        </span>
        <span />
      </div>
      <div
        class="item-group-container footer"
      >
        <span>
          View Profile
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
